pkg/tfbridge/serve.go

```diff
@@ -16,16 +16,52 @@ package tfbridge
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/pulumi/pulumi-terraform-bridge/v3/unstable/metadata"
+	"github.com/pulumi/pulumi-terraform-bridge/x/muxer"
 	"github.com/pulumi/pulumi/pkg/v3/resource/provider"
-	lumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
 )
 
 // Serve fires up a Pulumi resource provider listening to inbound gRPC traffic,
 // and translates calls from Pulumi into actions against the provided Terraform Provider.
-func Serve(module string, version string, info ProviderInfo, pulumiSchema []byte) error {
+func Serve(module string, version string, info ProviderInfo, pulumiSchema []byte, options ...Option) error {
+	opts := opts{}
+	for _, applyOption := range options {
+		applyOption(&opts)
+	}
+
 	// Create a new resource provider server and listen for and serve incoming connections.
-	return provider.Main(module, func(host *provider.HostClient) (lumirpc.ResourceProviderServer, error) {
+	return provider.Main(module, func(host *provider.HostClient) (pulumirpc.ResourceProviderServer, error) {
+		if len(opts.muxWith) > 0 {
```

Comment:
Nit: the glue code here can go into x/muxer.

Reply:
Can you explain how? The glue code here calls bridge functions like `metadata.Get` and `NewProvider`. Right now the muxer does not depend on the rest of the bridge, and I would like to keep it that way.